### PR TITLE
Enable running docker and podman in the same job for BCI tests

### PR DESCRIPTION
**Pull request by jlausuch**
Each BCI run will produce some log files, e.g. junit_base.xml.
But if we run the same command twice (using different engine)
those files will be overwritten.
With this PR, after each BCI test run, those files will be renamed
with the container engine.
I have createda third module to parse all the logs generated by the
tests.

https://progress.opensuse.org/issues/113501

VRs:
[base image 15-SP4](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=27.11.4_sles15-image&groupid=398)
[lang image](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=7.19_golang-1.18-image&groupid=398)

### CODE REVIEW

File: tests/containers/bci_logs.pm [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    assert_script_run("cd /root/BCI-tests");
+    my $dom = XML::LibXML::Document->new('1.0', 'utf-8');
+    my $root = $dom->createElement('testsuites');
+    $dom->setDocumentElement($root);
+    my $result_files = script_output('ls | grep junit');
```
Comment (pdostal) [inline tests/containers/bci_logs.pm:28]:
```suggestion
    my $result_files = script_output('find /root/BCI-tests -type f -name "*.junit"');
```
With this, you can also avoid `cd`.

File: tests/containers/bci_prepare.pm [hunk 2]
```diff
@@ -98,15 +99,17 @@ sub run {
...
     record_info('Clone', "Clone BCI tests repository: $bci_tests_repo");
     my $branch = $bci_tests_branch ? "-b $bci_tests_branch" : '';
-    assert_script_run("git clone $branch -q --depth 1 $bci_tests_repo");
+    assert_script_run("git clone $branch -q --depth 1 $bci_tests_repo /root/BCI-tests");
 
     # Pull the image in advance
     if (my $image = get_var('CONTAINER_IMAGE_TO_TEST')) {
         record_info('IMAGE', $image);
+        # If $engines are multiple (e.g. CONTAINER_RUNTIME=podman,docker), we just pick one of them for this check
+        my $engine = ($engines =~ /podman/) ? 'podman' : 'docker';
```
Comment (ilausuch) [inline tests/containers/bci_prepare.pm:112]: Would be better?
```
if ($engine =~ /podman/) { ... }
else if($engine =~ /docker/) {...}
else No valid engine

**Review by ilausuch — APPROVED** [review]

**Review by pdostal — APPROVED** [review]
This is 💯 nice to see!

File: tests/containers/bci_prepare.pm (hunk 2 above)
Comment (jlausuch) [reply tests/containers/bci_prepare.pm:112]: ok, why not

Comment (m-dati) [reply tests/containers/bci_prepare.pm:112]: if **$engines**="podman,docker", where will it span both values in order to repeat the test for each engine?

Comment (jlausuch) [reply tests/containers/bci_prepare.pm]: I have updated it. This module is executed only 1 time, I just pick one of the engines that are in the variable.

File: tests/containers/bci_logs.pm (hunk 1 above)
Comment (jlausuch) [reply tests/containers/bci_logs.pm]: I am renaming those files after bci_test.pm module, but the suggestion is still valid, thank you.

File: tests/containers/bci_prepare.pm (hunk 2 above)
Comment (m-dati) [reply tests/containers/bci_prepare.pm]: But _docker_ set the pulled repo in a different place than _podman_, so we should repeat the $engine pull for each type

Comment (jlausuch) [reply tests/containers/bci_prepare.pm]: Right, but this code block is just to verify the build tag of the image and report also the size to some database. The BCI-tests will pull the image themselves when needed.

File: tests/containers/bci_logs.pm (hunk 1 above)
Comment (jlausuch) [reply tests/containers/bci_logs.pm]: Updated with both suggestions. `find /root/BCI-tests -type f -name "junit_*.xml"`

**Comment by jlausuch** [comment]
Re-running VRs: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=27.11.4_sles15-image&groupid=398

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Pull request merged**